src/ngraph/pass/opset1_transform.cpp

```diff
@@ -0,0 +1,94 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "ngraph/pass/opset1_transform.hpp"
+#include "ngraph/graph_util.hpp"
+#include "ngraph/op/get_output_element.hpp"
+#include "ngraph/op/softmax.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+// @TODO: Shouldn't this be moved to a common utility class? This mapping to OP_TYPEID gets repeated many times.
+#define NGRAPH_OP(a, b) a,
+enum class OP_TYPEID
+{
+#include "ngraph/op/op_tbl.hpp"
+};
+#undef NGRAPH_OP
+
+#define NGRAPH_OP(a, b) {#a, OP_TYPEID::a},
+static unordered_map<string, OP_TYPEID> typeid_map{
+#include "ngraph/op/op_tbl.hpp"
+};
+#undef NGRAPH_OP
+
+static OP_TYPEID get_typeid(shared_ptr<Node> node)
+{
+    OP_TYPEID type_id;
+    auto it = typeid_map.find(node->description());
+    if (it != typeid_map.end())
+    {
+        type_id = it->second;
+    }
+    else
+    {
+        throw unsupported_op("Unsupported op '" + node->description() + "'");
+    }
+    return type_id;
+}
+// END mapping to OP_TYPEID
+
+bool pass::Opset1Transformation::run_on_node(shared_ptr<Node> node)
+{
+    bool modified = false;
+
+    size_t op_version = node->get_op_version();
+
+    if (op_version == 1)
+    {
+        return modified;
+    }
+
+    if (op_version != 0)
+    {
+        throw ngraph_error("Op version 1 transformation pass failed for " + node->get_name() +
+                           ", only op version 0 operations expected. Op version " +
+                           to_string(op_version) + " found.");
+    }
+
+    switch (get_typeid(node))
+    {
+    case OP_TYPEID::Softmax:
+    {
+        auto tmp = dynamic_cast<const op::v0::Softmax*>(node.get());
+        AxisSet axes = tmp->get_axes();
+
+        NGRAPH_CHECK(
+            axes.size() == 1,
+            "Unable to convert Softmax:0 to Softmax:1 with zero or more than one axis. Node: ",
```

Comment:
This is not complete transformation if we react this way. To have reliable and complete transformation the result should be Transpose(to gather dimensions in case if axis set is sparse) -> Reshape(to flatten several axis to one) -> v1::Softmax -> Reshape -> Transpose.

Reply:
Actually it should be even more complicated if we consider the ability to be reshaped for a different shape. In this case we cannot deduce target shape for Reshape op as a constant. Instead it should be calculated as a sub-graph that depends on ShapeOf(input_tensor) and contains such operations as Split or StridedSlice, Multiply and Concat. And only then this target shape is consumed as the second input into Reshape.

Reply:
@slyalin, @aprocter I agree with @postrational The support for that kind of transformation has not been implemented yet so we want to add it in separate PR

Reply:
@postrational @mbencer Agreed---let's do it that way. For example, #3536 is waiting on this.

Reply:
That's why v1::Reshape should be implemented first, then Softmax. There is a natural order in the new ops implementation process. But the tricky part is that we are discovering this order just in time when a particular op is being implemented. OK, let's postpone it right up to the moment when v1::Reshape is ready.

Reply:
Right- a lot of "oh, I see..." moments to come. Since subsequent PRs will not be blocking the underlying versioning infrastructure (in this case, Softmax was just chosen as a case study to try it out), doing things in a natural order should be less of a problem going forwards.